classes/ProductSale.php

```diff
@@ -234,29 +234,31 @@ public static function getBestSalesLight($idLang, $pageNumber = 0, $nbProducts =
      * Add Product sale.
      *
      * @param int $productId Product ID
+     * @param int $idShop Shop ID
      * @param int $qty Quantity
      *
      * @return bool Indicates whether the sale was successfully added
      */
-    public static function addProductSale($productId, $qty = 1)
+    public static function addProductSale($productId, $idShop, $qty = 1)
```

Comment:
As @kpodemski suggested this is a breaking change so instead you should have something like:

```suggestion
    public static function addProductSale($productId, $qty = 1, $shopId = null)
```

And when `$shopId` is null retrieve it from the shop context (be careful there can be multiple IDs if you are in a group or All shops context so you'll have to use a loop)

Reply:
Sorry for being late. I finally found the time to implement your recommendations.